### PR TITLE
Support to configure the cache size of vineyard fuse

### PR DESCRIPTION
**Pull request by dashanji**
### Ⅰ. Describe what this PR does

Part of https://github.com/fluid-cloudnative/fluid/pull/3749

### CODE REVIEW

File: charts/vineyard/templates/fuse/daemonset.yaml [hunk 1]
```diff
@@ -76,6 +76,8 @@ spec:
             value: {{ .Values.fuse.targetPath }}/rpc-conf
           - name: PRESTOP_MARKER
             value: /tmp/prestop-marker
+          - name: CACHE_SIZE
```
Comment (cheyang) [inline charts/vineyard/templates/fuse/daemonset.yaml:79]: How about using 

```yaml
{{- if .Values.fuse.options.cacheSize }}
- name: CACHE_SIZE
  value: "{{ .Values.fuse.options.cacheSize }}"
{{- end }}
```

File: api/v1alpha1/vineyardruntime_types.go [hunk 2]
```diff
@@ -174,6 +174,18 @@ type VineyardSockSpec struct {
...
+	// Supported options are as follows.
+	//   cache-size: the size of cache in vineyard fuse, the format could be
+	//               1024M, 1024000, 1G, or 1Gi.
+	//               Default is "0", which means no cache.
+	// Default is as follows.
+	// fuse:
+	//   options:
+	//     cache-size: "0"
+	// +optional
+	Options map[string]string `json:"options,omitempty"`
```
Comment (cheyang) [inline api/v1alpha1/vineyardruntime_types.go:188]: How to handle other options? I think they are the parts of command and args.

File: api/v1alpha1/vineyardruntime_types.go [hunk 3]
```diff
@@ -174,6 +174,18 @@ type VineyardSockSpec struct {
...
 }
```
Comment (cheyang) [inline api/v1alpha1/vineyardruntime_types.go:189]: I also believe that the name VineyardClientSocketSpec would be more appropriate.

**Review by cheyang — CHANGES_REQUESTED** [review]

File: api/v1alpha1/vineyardruntime_types.go (hunk 2 above)
Comment (dashanji) [reply api/v1alpha1/vineyardruntime_types.go:233]: Thanks for the advice. I have used a range to iterate all fuse options.

File: api/v1alpha1/vineyardruntime_types.go (hunk 3 above)
Comment (dashanji) [reply api/v1alpha1/vineyardruntime_types.go:234]: Done.

File: charts/vineyard/templates/fuse/daemonset.yaml (hunk 1 above)
Comment (dashanji) [reply charts/vineyard/templates/fuse/daemonset.yaml]: I used a range to avoid specifying each argument one by one.

**Comment by cheyang** [comment]
@TrafalgarZZZ Please also take a look. Thanks.

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**